### DISCUSSION TITLE
Use _id with own string

### DISCUSSION THREAD

**Discussion by MadEddieFFS** [colinhacks/zod#2152] [Q&A] [answered]
Sorry to ask here but wondered how to setup my User object with my own string as the _id using TS:
**users.model.ts**
```ts
import { db } from './../../../db';
import * as z from 'zod';
import { WithId } from 'mongodb';

export const User = z.object({
    email: z.string().email(),
    forename: z.string().min(3),
    surname: z.string().min(3),
    picture: z.string(),
    createdAt: z.date(),
    updatedAt: z.date(),
    _id: z.string(),
    role: z.string().default('user')
});

export type User = z.infer<typeof User>;
export type UserWithId = WithId<User>;
export const Users = db.collection<User>('users');
```
Seems fine but when I use FindOne on the _id field it wants an ObjectId

**user.controller.ts**
```ts
import { User, Users, UserWithId } from "./users.model";
const user = await Users.findOne({ _id: new ObjectId(req.params.id) });
```
Which obviously it can't do to a rrq.param.id string like "MyUserIDIsAString"

Any help appreciated

**Comment by JacobWeisenburger** [comment, marked as answer]
This doesn't seem like a problem with zod. This seems like a problem with mongodb/mongoose. Here is probably a better place to ask this question: https://github.com/Automattic/mongoose/discussions

**Reply by MadEddieFFS** [reply]
Your are quite correct :(
